.circleci/config.yml

```diff
@@ -419,13 +420,17 @@ jobs:
           command: |
             rm -rf sql/
             cp -r /tmp/workspace/private-generated-sql/sql sql
-      - gcp-gcr/gcr-auth
-      - gcp-gcr/build-image:
-          image: bigquery-etl
-          tag: ${CIRCLE_TAG:-latest}
-      - gcp-gcr/push-image:
+      - when:
+          condition: *deploy-condition
+          steps:
+            - gcp-gcr/gcr-auth
```

Comment:
Can this gcr-auth step be delayed until after `build-image` so that it can be nested under the same `when` condition?

Reply:
this follows the pattern set in the [gcp-gcr orb job](https://github.com/CircleCI-Public/gcp-gcr-orb/blob/master/src/jobs/build-and-push-image.yml#L123-L147) and [docker orb job](https://github.com/CircleCI-Public/docker-orb/blob/master/src/jobs/publish.yml#L165-L200) of doing auth->build->push.

At least one reason for doing this could be that docker has a `--cache-from` flag now that allows one to use the previous image layers as a build cache: https://github.com/CircleCI-Public/docker-orb/blob/master/src/commands/build.yml#L128